### PR TITLE
Update bower deps to latest versions

### PR DESCRIPTION
**Pull request by charlescrain**
I found that argonaut and some other libs were behind and causing conflict when adding this library to newer projects

### CODE REVIEW

**Comment by charlescrain** [comment]
@safareli @blinky3713 Let me know if there's something i missed 😄

**Comment by martyall** [comment]
looks good to me!

File: bower.json [hunk 1]
```diff
@@ -18,14 +18,13 @@
     "output"
   ],
   "dependencies": {
-    "purescript-prelude": "^4.0.0",
+    "purescript-prelude": "^4.1.0",
     "purescript-ring-modules": "^5.0.1",
-    "purescript-foreign-generic": "^7.0.0",
-    "purescript-simple-json": "^4.0.0",
-    "purescript-ordered-collections": "^1.0.0",
+    "purescript-foreign-generic": "^8.0.0",
```
Comment (safareli) [inline bower.json:23]: this is main change here, and as the change in `purescript-foreign-generic` is braking if this lib was used with `purescript-foreign-generic@7` releasing this change with patch version will result in build error, therefore it's braking and we should bump major.

File: bower.json [hunk 2]
```diff
@@ -18,14 +18,13 @@
...
+    "purescript-simple-json": "^4.3.0",
```
Comment (safareli) [inline bower.json:24]: We do not really need to update number in minor or patch part of version as we use `^`, `^x.a.b` means any version which has x as major and anything more then a.b is fine. if we used a new function which is added to purescript-simple-json 4.3 for example, which we don't.

Also this change is somewhat braking. This lib works fine with purescript-simple-json v4 as well as v4.1, v4.2, v4.3 etc, if this lib is installed in some lib/app where exact v4 of purescript-simple-json is used as dependency, build of that would fail. 

(Changing version of dependency is mostly a braking change.)

Basically we don't need to make change minor or patch part of versions where ^ is used unless the lib where we made major bump needs it (for example purescript-foreign-generic needs prelude ^4.1)

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "purescript-eth-core",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (safareli) [inline package.json:3]: We are actually at version  v3.0.0 and next one will be  v4.0.0, so please either revert this change and I will remove it later, or apply this change to it:

```
 {
   "name": "purescript-eth-core",
-  "version": "1.0.0",
-  "description": "",
-  "main": "index.js",
+  "private": true,
   "dependencies": {
     "bn.js": "^4.11.0",
     "keccak": "^1.0.2",
@@ -16,7 +14,5 @@
   },
   "scripts": {
     "test": "pulp test"
-  },
-  "author": "",
-  "license": "ISC"
+  }
 }
```

In PS libs we use package.json to only get npm deps and sometimes add `scripts` to it. versions of lib are managed using git tags https://github.com/f-o-a-m/purescript-eth-core/releases

**Review by safareli — CHANGES_REQUESTED** [review]
Basically we need to make this change to bower.json:

```difff
   "dependencies": {
-    "purescript-prelude": "^4.0.0",
+    "purescript-prelude": "^4.1.0",
     "purescript-ring-modules": "^5.0.1",
-    "purescript-foreign-generic": "^7.0.0",
+    "purescript-foreign-generic": "^8.0.0",
     "purescript-simple-json": "^4.0.0",
     "purescript-ordered-collections": "^1.0.0",
     "purescript-bytestrings": "^7.0.0",
-    "purescript-simple-json": "^4.0.0",
-    "purescript-argonaut": "^4.0.0"
+    "purescript-argonaut": "^5.0.0"
   },
```

(unless there is some specific reason to bump simple-json or ordered-collections)

**Comment by safareli** [comment]
Also note before we merge this, all libs which depend on it (ps-web3*) should also be updated, unless you use this lib separately.

**Comment by martyall** [comment]
closing in favor of #19
